x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunkingSettings.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.chunking;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.inference.ChunkingSettings;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class SentenceBoundaryChunkingSettings extends ChunkingSettings {
+    public static final String NAME = "SentenceBoundaryChunkingSettings";
+    protected static final String STRATEGY = ChunkingStrategy.SENTENCE.toString();
+    protected final int maxChunkSize;
+
+    public SentenceBoundaryChunkingSettings(Integer maxChunkSize) {
+        super(STRATEGY);
+        this.maxChunkSize = maxChunkSize;
+    }
+
+    public SentenceBoundaryChunkingSettings(StreamInput in) throws IOException {
+        super(STRATEGY);
```

Comment:
The abstract base class could define a `public ChunkingStrategy getStrategy()` method which is implemented in these classes as a hard coded return value. I'm making this suggestion because the base class does not do anything with the strategy, it does not serialise the value or write it in `toXContent()`.

Reply:
Yes, with the comment above regarding changing the base class's `chunkingStrategy` to a `ChunkingStrategy` value, I can update the `getStrategy` function int the base class to return a `ChunkingStrategy` object.

Reply:
Actually maybe I've misunderstood your comment. Was the suggestion to update the returned type of the existing `ChunkingStrategy` to avoid conversions or is the suggestion to remove the strategy from the abstract base class as a whole to simplify it by adding a `getChunkingStrategy` method that each `ChunkingStrategy` class overrides?

Reply:
Change `getChunkingStrategy()` to return the `ChunkingStrategy ` enum instead of a string.  

My other suggestion is a matter of style you can feel free to ignore... what I was trying to say is that `getChunkingStrategy()` can be abstract in the ABC, in this class you are using a static constant `protected static final String STRATEGY` you don't need the static member if you are passing it to the base class ctor. Instead just implement `getChunkingStrategy()` as `return ChunkingStrategy.SENTENCE;` and remove the static constant.

Reply:
Thanks for clarifying. I think it makes sense to move this out of the abstract base class and into the specific implementations. This simplifies the base class and even lets us convert it to just an interface. I'll make that change.